master: `panic: 15.2 is not in dotted-tri format`

```console
$ limactl start
? Creating an instance "default" Proceed with the current configuration
panic: 15.2 is not in dotted-tri format

goroutine 1 [running]:
github.com/coreos/go-semver/semver.Must(...)
        /Users/suda/gopath/pkg/mod/github.com/coreos/go-semver@v0.3.1/semver/semver.go:65
github.com/coreos/go-semver/semver.New({0x101298059?, 0x10?})
        /Users/suda/gopath/pkg/mod/github.com/coreos/go-semver@v0.3.1/semver/semver.go:49 +0x44
github.com/lima-vm/lima/pkg/limayaml.HasHostCPU()
        /Users/suda/gopath/src/github.com/lima-vm/lima/pkg/limayaml/defaults.go:1198 +0x5c
github.com/lima-vm/lima/pkg/limayaml.defaultCPUType()
        /Users/suda/gopath/src/github.com/lima-vm/lima/pkg/limayaml/defaults.go:67 +0x1e4
github.com/lima-vm/lima/pkg/limayaml.FillDefault(0x140004d0dc8, 0x140004d1088, 0x140004d1348, {0x140004fc4e0, 0x23}, 0x1)
        /Users/suda/gopath/src/github.com/lima-vm/lima/pkg/limayaml/defaults.go:292 +0xce4
github.com/lima-vm/lima/pkg/limayaml.load({0x14000580000, 0x6b63, 0x6b64}, {0x140004fc4e0, 0x23}, 0x1)
        /Users/suda/gopath/src/github.com/lima-vm/lima/pkg/limayaml/load.go:66 +0x3e4
github.com/lima-vm/lima/pkg/limayaml.LoadWithWarnings(...)
        /Users/suda/gopath/src/github.com/lima-vm/lima/pkg/limayaml/load.go:25
github.com/lima-vm/lima/pkg/instance.Create({0x1017c14c0, 0x1026e0b80}, {0x10129e7c6, 0x7}, {0x14000580000, 0x6b63, 0x6b64}, 0x1)
        /Users/suda/gopath/src/github.com/lima-vm/lima/pkg/instance/create.go:44 +0x144
main.loadOrCreateInstance(0x140004b9208, {0x1026e0b80?, 0x0?, 0x0?}, 0x0)
        /Users/suda/gopath/src/github.com/lima-vm/lima/cmd/limactl/start.go:224 +0xbac
main.startAction(0x140004b9208, {0x1026e0b80, 0x0, 0x0})
        /Users/suda/gopath/src/github.com/lima-vm/lima/cmd/limactl/start.go:406 +0x50
github.com/spf13/cobra.(*Command).execute(0x140004b9208, {0x1026e0b80, 0x0, 0x0})
        /Users/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0x834
github.com/spf13/cobra.(*Command).ExecuteC(0x140004b8c08)
        /Users/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/suda/gopath/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
        /Users/suda/gopath/src/github.com/lima-vm/lima/cmd/limactl/main.go:28 +0x20
```

Tested on macOS 15.3 on Apple M4 Max.

Regression in:
- https://github.com/lima-vm/lima/pull/3197

--- a/pkg/limayaml/defaults.go
+++ b/pkg/limayaml/defaults.go
@@ -1195,7 +1195,7 @@ func HasHostCPU() bool {
 	case "darwin":
 		if hasSMEDarwin() {
 			macOSProductVersion, err := osutil.ProductVersion()
-			if err != nil || macOSProductVersion.Equal(*semver.New("15.2")) {
+			if err != nil || (macOSProductVersion.Major == 15 && macOSProductVersion.Minor == 2) {
 				// SME is available since Apple M4 running macOS 15.2, but it was broken on macOS 15.2.
 				// It has been fixed in 15.3.
 				//

